<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using-AraNetBench â€¢ AraNetBench</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using-AraNetBench">
<meta property="og:description" content="AraNetBench">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">AraNetBench</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Using-AraNetBench.html">Using-AraNetBench</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Using-AraNetBench_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using-AraNetBench</h1>
            
      
      
      <div class="hidden name"><code>Using-AraNetBench.Rmd</code></div>

    </div>

    
    
<div id="principle" class="section level1">
<h1 class="hasAnchor">
<a href="#principle" class="anchor"></a>Principle</h1>
<blockquote>
<p>AraNetBench is a package designed to evaluate the result of a network inference method against on known, experimentally determined, regulatory interactions in <em>Arabidopsis thaliana</em>.</p>
</blockquote>
<p>In systems biology, reconstructing gene regulatory networks is at the same time an ultimate goal, as well as a real challenge. Usually, network inference methods predict connections between genes using expression values in a set of experimental conditions and genes of interest. Those predicted connections are supposed to carry information about the transcriptional dependencies at stake in the response to a developmental of adaptive mechanism.</p>
<p>However, biological systems and the true regulatory networks orchestrating gene expression changes in cells are still far from being fully mapped. This makes it very difficult to asses weather network inference methods correctly predict those regulatory relationships between genes.</p>
<p>AraNetBench was made in an effort quantitatively validate inferred networks in <em>Arabidopsis thaliana</em> against the <a href="https://connectf.org/">connecTF database</a>. Although the connecTF plateform is a great online graphical tool, is does not allow to streamline and generalize the validation processes like scripts or automated pipelines would offer.</p>
<p>This package thus contains several functions that can be called from R scripts given any inferred network in the form of an edges dataframe. Those functions :</p>
<ul>
<li><p>Compute <strong>evaluation metrics</strong> on an inferred network such as true positives and true positive rate (interactions that are supported by experimental evidence), or false positives and false positive rate.</p></li>
<li><p><strong>Display a view of the network</strong>, with its edges colored depending on their validation status, and the nature of the supporting information in connecTF.</p></li>
<li><p>Asses if the inference allowed a <strong>significant improvement in biological meaning over randomness</strong>.</p></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AraNetBench</span><span class="op">)</span></code></pre></div>
</div>
<div id="loading-an-example-network" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-an-example-network" class="anchor"></a>Loading an example network</h1>
<p>AraNetBench comes with its companion dataset, i.e an infered network of genes responsive to heat under manitol stress. The data was collected by <a href="https://pubmed.ncbi.nlm.nih.gov/32442250/">Sewelam et Al. Molecular plant responses to combined abiotic stresses put a spotlight on unknown and abundant genes. J Exp Bot. 2020.</a>, and a network on those genes was inferred in <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-021-07659-2">the article</a> presenting <a href="https://oceanecsn.github.io/DIANE/">DIANE</a>.</p>
<p>Here is an overview of this infered network :</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"abiotic_stress_Heat_genes_net"</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">abiotic_stress_Heat_genes_net</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="92%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th align="left">from</th>
<th align="left">to</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">AT5G43840</td>
<td align="left">AT1G52690</td>
</tr>
<tr class="even">
<td align="left">AT5G43840</td>
<td align="left">AT4G17470</td>
</tr>
<tr class="odd">
<td align="left">AT1G66390</td>
<td align="left">AT4G08570</td>
</tr>
<tr class="even">
<td align="left">mean_AT1G01060-AT3G09600-AT5G24110-AT3G12910-AT5G56960-AT1G20693-AT2G37430</td>
<td align="left">AT3G22235</td>
</tr>
<tr class="odd">
<td align="left">mean_AT1G17460-AT5G57150-AT5G60100-AT5G48250-AT3G62090-AT1G09530-AT5G63160-AT1G76590-AT3G11020-AT1G11100-AT1G67370-AT3G14740</td>
<td align="left">AT1G60190</td>
</tr>
<tr class="even">
<td align="left">AT1G44830</td>
<td align="left">AT5G65390</td>
</tr>
</tbody>
</table>
<p>All input networks should be formatted identically (i.e a column <code>from</code>, and a column <code>to</code>, both containing gene AGI). If the inferred network comes from DIANE (not a prerequiste), grouped regulators will be temporarily ungrouped to compute evaluation metrics, then regrouped for visualisation.</p>
</div>
<div id="overview-of-validation-data" class="section level1">
<h1 class="hasAnchor">
<a href="#overview-of-validation-data" class="anchor"></a>Overview of validation data</h1>
<p>The validation database contains different experiment types to confirm predicted interactions :</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"validated_edges"</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">validated_edges</span><span class="op">$</span><span class="va">type</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">CHIPSeq</td>
<td align="right">206974</td>
</tr>
<tr class="even">
<td align="left">DAPSeq</td>
<td align="right">1983037</td>
</tr>
<tr class="odd">
<td align="left">Litterature</td>
<td align="right">1431</td>
</tr>
<tr class="even">
<td align="left">TARGET</td>
<td align="right">85144</td>
</tr>
</tbody>
</table>
<p>By default, DAPSeq, CHIPSeq and TARGET data are used. This can by altered with the <code>validation</code> argument, containing the desired nature of the validation links.</p>
</div>
<div id="validating-a-network" class="section level1">
<h1 class="hasAnchor">
<a href="#validating-a-network" class="anchor"></a>Validating a network</h1>
<p>Call <code>evaluate_network</code> to compute evaluation metrics on an inferred network such as true positives and true positive rate (interactions that are supported by experimental evidence), or false positives and false positive rate. It will also return the same network as the input, but with an additional column about the supporting information on each edge.</p>
<p>For reproducible results, set the seed before random networks perumtations :</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">999</span><span class="op">)</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate_network.html">evaluate_network</a></span><span class="op">(</span><span class="va">abiotic_stress_Heat_genes_net</span><span class="op">)</span>
<span class="va">results</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"tp"</span>, <span class="st">"fp"</span>, <span class="st">"tpr"</span>, <span class="st">"fpr"</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; $tp</span>
<span class="co">#&gt; [1] 266</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $fp</span>
<span class="co">#&gt; [1] 708</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $tpr</span>
<span class="co">#&gt; [1] 0.2731006</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $fpr</span>
<span class="co">#&gt; [1] 0.7268994</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">edges</span>, n <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">from</th>
<th align="left">to</th>
<th align="left">type</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">27</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT5G48657</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="even">
<td align="left">29</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT5G46050</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="odd">
<td align="left">30</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT3G46110</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="even">
<td align="left">34</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT1G66880</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="odd">
<td align="left">39</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT4G14365</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="even">
<td align="left">41</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT2G35980</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="odd">
<td align="left">46</td>
<td align="left">AT5G43840</td>
<td align="left">AT4G02280</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="even">
<td align="left">141</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT5G57010</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="odd">
<td align="left">150</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT1G78860</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="even">
<td align="left">153</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT2G37980</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="odd">
<td align="left">194</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT4G37900</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="even">
<td align="left">207</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT3G15356</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="odd">
<td align="left">208</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT5G19240</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="even">
<td align="left">226</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT4G21903</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="odd">
<td align="left">297</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT2G02930</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="even">
<td align="left">339</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT1G62300</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="odd">
<td align="left">405</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT2G39210</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="even">
<td align="left">412</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT1G30720</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="odd">
<td align="left">413</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT3G19930</td>
<td align="left">CHIPSeq</td>
</tr>
<tr class="even">
<td align="left">455</td>
<td align="left">mean_AT5G03740-AT1G56170</td>
<td align="left">AT1G35230</td>
<td align="left">CHIPSeq</td>
</tr>
</tbody>
</table>
</div>
<div id="drawing-the-result-of-a-network-evaluation" class="section level1">
<h1 class="hasAnchor">
<a href="#drawing-the-result-of-a-network-evaluation" class="anchor"></a>Drawing the result of a network evaluation</h1>
<p>To display a view of the network, with its edges colored depending on their validation status, and the nature of the supporting information in connecTF, call <code>draw_evaluated_network</code> on the result of <code>evaluate_network</code> :</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/draw_evaluated_network.html">draw_evaluated_network</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></code></pre></div>
<p><img src="Using-AraNetBench_files/figure-html/unnamed-chunk-5-1.png" width="768"></p>
<p>To save the output in a file :</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/draw_evaluated_network.html">draw_evaluated_network</a></span><span class="op">(</span><span class="va">results</span>, export <span class="op">=</span> <span class="cn">TRUE</span>, filename <span class="op">=</span> <span class="st">"evaluated_network.png"</span><span class="op">)</span></code></pre></div>
</div>
<div id="permutation-test-on-an-inferred-network" class="section level1">
<h1 class="hasAnchor">
<a href="#permutation-test-on-an-inferred-network" class="anchor"></a>Permutation test on an inferred network</h1>
<p>To test if the network contains biological signal or if the rate of validated edges does not differ from what is expected under random inference, run <code>test_validation_rate</code> on the inferred network.</p>
<p>For example, DAPSeq, CHIPSeq and TARGET information clearly seem to be captured by the network inference in the demo dataset:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/test_validation_rate.html">test_validation_rate</a></span><span class="op">(</span><span class="va">abiotic_stress_Heat_genes_net</span>, N <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<p><img src="Using-AraNetBench_files/figure-html/unnamed-chunk-7-1.png" width="768"></p>
<p>Whereas CHIPSeq signal alone in the network could have been achieved by randomly linking TFs to targetsâ€¦</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/test_validation_rate.html">test_validation_rate</a></span><span class="op">(</span><span class="va">abiotic_stress_Heat_genes_net</span>, validation <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CHIPSeq"</span><span class="op">)</span>, N <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<p><img src="Using-AraNetBench_files/figure-html/unnamed-chunk-8-1.png" width="768"></p>
<p>Other functions to come soon, like drawing ROC curves on fully connected weighted inferred networks!</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Oceane Cassan.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
